conda/shell/etc/fish/conf.d/conda.fish

```diff
@@ -87,11 +87,13 @@ end
 
 # Faster but less tested (?)
 function __fish_conda_commands
-  command echo -e "activate\ndeactivate" ;and ls --color=none (conda info --root)/bin/conda-* | sed -r 's/^.*conda-([a-z]+)/\1/'
+  string replace -r '.*_([a-z]+)\.py$' '$1' $_CONDA_ROOT/lib/python*/site-packages/conda/cli/main_*.py
+  echo activate
+  echo deactivate
 end
 
 function __fish_conda_envs
-  command echo root ;and ls -1 --color=none (conda info --root)/envs/
+  basename -a $_CONDA_ROOT/envs/*/
```

Comment:
Better here might be using the output of `conda info --envs`.  Not sure how to do that though.  Maybe something like

    command conda info --envs | awk '{print $1}'

?

Reply:
I used `conda info --envs | awk 'NR>2{print $1}' ` at some point but switched to the `envs` listing due to performance issues (but only philosophically maybe, because `conda info --root` is equally slow). Maybe in 4.4 now that's not a problem and that way we can list additional envs not present in the default directory.

Reply:
OK, maybe we should leave it just to this PR for 4.4.0, and then you all can play with this again after you get a feel for things with 4.4.